core/src/test/java/org/elasticsearch/cluster/ClusterLimitIT.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.containsString;
+
+public class ClusterLimitIT extends ESIntegTestCase {
+    public void testInternalActionsAreNotSizeLimited() throws Exception {
+        // keep default circuit breaker limit until the cluster is up as the test framework also issues client requests in this phase
+        ensureGreen();
+        int clusterSize = internalCluster().size();
+
+        Settings settings = Settings.builder()
+            .put(HierarchyCircuitBreakerService.IN_FLIGHT_REQUESTS_CIRCUIT_BREAKER_LIMIT_SETTING.getKey(), new ByteSizeValue(0))
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setTransientSettings(settings));
+
+        try {
+            internalCluster().transportClient().admin().cluster().prepareNodesInfo().get();
+            fail("Should have thrown a CircuitBreakingException");
+        } catch (CircuitBreakingException ex) {
+            assertThat(ex.getMessage(), containsString("data for [<transport_request>] would be larger than limit of [0/0b]"));
+        }
+        // give the cluster a chance to perform requests
+        Thread.sleep(randomIntBetween(1_000, 9_000));
```

Comment:
Can we do anything other than sleep? Sleeping doesn't give us much in the way of guarantees.